dev/com.ibm.ws.security.fat.common/src/com/ibm/ws/security/fat/common/actions/SecurityTestFeatureEE10RepeatAction.java

```diff
@@ -0,0 +1,136 @@
+/*******************************************************************************
+ * Copyright (c) 2020, 2022 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ * IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.security.fat.common.actions;
+
+import java.util.Set;
+
+import com.ibm.websphere.simplicity.Machine;
+import com.ibm.websphere.simplicity.OperatingSystem;
+import com.ibm.websphere.simplicity.log.Log;
+
+import componenttest.custom.junit.runner.Mode.TestMode;
+import componenttest.custom.junit.runner.TestModeFilter;
+import componenttest.rules.repeater.JakartaEE10Action;
+
+public class SecurityTestFeatureEE10RepeatAction extends JakartaEE10Action {
+
+    public static Class<?> thisClass = SecurityTestFeatureEE10RepeatAction.class;
+
+    protected String complexId = JakartaEE10Action.ID;
+    private TestMode testRunMode = TestModeFilter.FRAMEWORK_TEST_MODE;
+    private boolean notAllowedOnWindows = false;
+
+    public SecurityTestFeatureEE10RepeatAction() {
+
+        complexId = JakartaEE10Action.ID;
+        Log.info(thisClass, "instance", complexId);
+        testRunMode = TestModeFilter.FRAMEWORK_TEST_MODE;
+        notAllowedOnWindows = false;
+        withID(complexId);
+    }
+
+    public SecurityTestFeatureEE10RepeatAction(String inNameExtension) {
+
+        complexId = JakartaEE10Action.ID + "_" + inNameExtension;
+        Log.info(thisClass, "instance", complexId);
+        testRunMode = TestModeFilter.FRAMEWORK_TEST_MODE;
+        notAllowedOnWindows = false;
+        withID(complexId);
+    }
+
+    @Override
+    public SecurityTestFeatureEE10RepeatAction withID(String id) {
+        return (SecurityTestFeatureEE10RepeatAction) super.withID(id);
+    }
+
+    @Override
+    public boolean isEnabled() {
+        Log.info(thisClass, "isEnabled", "testRunMode: " + testRunMode);
+        Log.info(thisClass, "isEnabled", "complexId: " + complexId);
+        Log.info(thisClass, "isEnabled", "Overall test mode: " + TestModeFilter.FRAMEWORK_TEST_MODE);
+        // allow if mode matches or mode not set
+        if (testRunMode != null && (TestModeFilter.FRAMEWORK_TEST_MODE != testRunMode)) {
+            Log.info(thisClass, "isEnabled", "Skipping action '" + toString() + "' because the test mode " + testRunMode +
+                    " is not valid for current mode " + TestModeFilter.FRAMEWORK_TEST_MODE);
+            return false;
+        }
```

Comment:
This is not the normal check for whether a test should run in the given mode.

Looking at the places `SecurityTestFeatureEE9RepeatAction.liteFATOnly()` is used though, it appears to be deliberate?

Reply:
It is close to what we do in the other code.  There is already lite FAT only option in the main code as well and could be change to use that function instead.  I just copied the EE9 implementation without changes.

Reply:
Just seems odd that we have that option at all anywhere, given that FULL mode is supposed to be a superset of LITE.

Reply:
The main problem is the slow build machines where running in full mode will cause things to go over 3 hours.  So some teams have chosen to run a subset of tests in lite mode only.  Likely this whole class could be removed and replaced with the main classes with existing function, but would have to investigate it.  I think that may be outside of the scope of this PR.